test/visual/review.js

```diff
@@ -26,15 +26,48 @@ const getHash = (context) => {
     return md5.digest('hex');
 };
 
+const getAzureUrl = (combination) => {
+    const baseUrl = 'https://swcpreviews.z13.web.core.windows.net';
+
+    // For main branch beta deployment
+    if (branch === 'main') {
+        return `${baseUrl}/beta/docs/`;
+    }
+
+    // Extract PR number using the same logic as CircleCI
+    let prNumber = null;
+
+    // Check environment variables (same as CircleCI)
+    if (process.env.CIRCLE_PULL_REQUEST) {
+        const match = process.env.CIRCLE_PULL_REQUEST.match(/\/pull\/(\d+)/);
+        if (match) {
+            prNumber = match[1];
+        }
+    } else if (process.env.CIRCLE_PR_NUMBER) {
+        prNumber = process.env.CIRCLE_PR_NUMBER;
+    } else if (branch.match(/^pull\/(\d+)$/)) {
+        // Forked PR branch pattern
+        const match = branch.match(/^pull\/(\d+)$/);
+        if (match) {
+            prNumber = match[1];
+        }
+    }
+
+    // For PR deployments
+    if (prNumber) {
+        return `${baseUrl}/pr-${prNumber}/${combination}/`;
+    }
+
+    // Default fallback for non-PR branches
+    return `${baseUrl}/${slugify(branch)}/${combination}/`;
+};
+
 const vrts = [];
 const themes = ['Spectrum', 'Express', 'Spectrum-two'];
 const scales = ['Medium', 'Large'];
 const colors = ['Lightest', 'Light', 'Dark', 'Darkest'];
 const directions = ['LTR', 'RTL'];
-vrts.push([
-    `High Contrast Mode | Medium | LTR`,
-    `https://${getHash(`${branch}-hcm`)}--spectrum-wc.netlify.app/review/`,
-]);
+vrts.push([`High Contrast Mode | Medium | LTR`, `${getAzureUrl('hcm')}`]);
 themes.map((theme) =>
```

Comment:
nit: I just noticed we're mapping but we could be `forEach`'ing everywhere.

Reply:
not sure if this is a :nit but a great callout. We are not using the return values from the `map` function but creating a side-effect. `foreach` works best and readable.